internal/pkg/otel/run.go

```diff
@@ -0,0 +1,121 @@
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+const buildDescription = "Elastic opentelemetry-collector distribution"
+
+func IsOtelConfig(ctx context.Context, pathConfigFile string) (bool, error) {
+	fileName := filepath.Base(pathConfigFile)
+	cleanFileName := strings.TrimSpace(strings.ToLower(strings.TrimSuffix(fileName, filepath.Ext(fileName))))
+	if cleanFileName == "otel" || cleanFileName == "otlp" || cleanFileName == "otelcol" {
+		return true, nil
+	}
+
+	rawConfig, err := config.LoadFile(pathConfigFile)
+	if err != nil {
+		return false, errors.New(err,
+			fmt.Sprintf("could not read configuration file %s", pathConfigFile),
+			errors.TypeFilesystem,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	mapConfig, err := rawConfig.ToMapStr()
+	if err != nil {
+		return false, errors.New(err,
+			errors.TypeConfig,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	// contains agent definition
+	_, hasAgent := mapConfig["agent"]
+	_, hasInputs := mapConfig["inputs"]
+	_, hasOutputs := mapConfig["outputs"]
+
+	if hasAgent || hasInputs || hasOutputs {
+		return false, nil
+	}
+
+	// contains otel service definition
+	_, hasService := mapConfig["service"]
+	_, hasExporters := mapConfig["exporters"]
+	_, hasReceivers := mapConfig["receivers"]
+	_, hasProcessors := mapConfig["processors"]
+
+	if hasService && (hasExporters || hasReceivers || hasProcessors) {
+		return true, nil
+	}
+
+	// default behavior is Elastic Agent
+	return false, nil
+}
+
+func Run(ctx context.Context, cancel context.CancelFunc, stop chan bool, testingMode bool) error {
+	fmt.Fprintln(os.Stdout, "Starting in otel mode")
+	settings, err := newSettings([]string{paths.ConfigFile()}, release.Version())
+	if err != nil {
+		return err
+	}
+
+	svc, err := otelcol.NewCollector(*settings)
+	if err != nil {
+		return err
+	}
+
+	// cancel context on stop from event manager
+	cancelCtx, cancel := context.WithCancel(ctx)
+	go func() {
+		<-stop
+		cancel()
+	}()
+	defer cancel()
+
+	return svc.Run(cancelCtx)
+}
+
+func newSettings(configPaths []string, version string) (*otelcol.CollectorSettings, error) {
+	buildInfo := component.BuildInfo{
+		Command:     os.Args[0],
+		Description: buildDescription,
+		Version:     version,
+	}
+
+	fmp := NewFileProviderWithDefaults()
+	configProviderSettings := otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs:       configPaths,
+			Providers:  map[string]confmap.Provider{fmp.Scheme(): fmp},
+			Converters: []confmap.Converter{expandconverter.New()},
+		},
+	}
+	provider, err := otelcol.NewConfigProvider(configProviderSettings)
+	if err != nil {
+		return nil, err
+	}
+
+	return &otelcol.CollectorSettings{
+		Factories:               components,
+		BuildInfo:               buildInfo,
+		ConfigProvider:          provider,
+		DisableGracefulShutdown: true,
```

Comment:
Probably worth commenting here that we're disabling graceful shutdown as we're handling that via the `cancelCtx` being passed to the collector's `Run` method in the `Run` function above?